### PR TITLE
Fixing update-testcases and adding the thus missing test.

### PR DESCRIPTION
**Pull request by mithro**

### CODE REVIEW

File: test/update-testcases.py [hunk 1]
```diff
@@ -54,7 +54,8 @@ def is_testcase_file(filename):
     for line in lines:
         sys.stdout.write(line)
 
-    sys.exit(not lines)
+    sys.exit(int(len(list(lines)) > 0))
```
Comment (dstoc) [inline test/update-testcases.py:57]: Can len return something which is not an integer?

**Comment by dstoc** [comment]
lgtm

File: test/update-testcases.py (hunk 1 above)
Comment (mithro) [inline test/update-testcases.py:57]: It can throw an exception is lines is not something which a **len** method...

**Comment by mithro** [comment]
Can you merge?

File: test/update-testcases.py (hunk 1 above)
Comment (dstoc) [inline test/update-testcases.py:57]: No need to call int() then?

Comment (mithro) [inline test/update-testcases.py:57]: sys.exit needs an int, len(list(lines)) > 0 returns a boolean..

**Pull request merged**